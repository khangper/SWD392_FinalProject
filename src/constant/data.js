export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
