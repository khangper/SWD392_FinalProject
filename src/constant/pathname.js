export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
