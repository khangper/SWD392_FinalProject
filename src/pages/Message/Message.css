.message-icon-msg {
  width: 20px;
  height: 20px;
  color: #333;
}

.message-icon {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-end;
  padding: 0 0 2.8px;
}

.message-title {
  position: relative;
  line-height: 26px;
  font-weight: 550;
  color: #333;
  font-size: 18px;
  margin-bottom: 0 !important;
}

.message-titles,
.message-label {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
}

.message-titles {
  align-items: flex-end;
  gap: 10px;
}

.message-label {
  align-items: flex-start;
}

.message-icon-search {
  width: 20px;
  height: 20px;
  color: rgb(194, 192, 192);
}

.message-button-search {
  padding: 8px 5px 5px 18px;
  border: 0;
  background-color: transparent;
}

.message-search-child,
.message-box-input-search,
.message-search {
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
}

.message-search-child {
  height: 30px;
  padding: 18.3px;
  font-size: 14px;
  flex-direction: column;
  background-color: transparent;
  width: 100%;
  border: 0;
  outline: 0;
}

.message-search-child::placeholder {
  color: rgb(194, 192, 192);
}

.message-box-input-search,
.message-search {
  flex-direction: row;
}

.message-box-input-search {
  flex: 1;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #e5e5e5;
}

.message-box-input-search:hover {
  border: 1px solid #333;
  color: #333;
}

.message-search {
  align-self: stretch;
  padding: 17px 30px 17px 17px;
}

.message-section {
  padding: 10px 10px 30px 15px;
}

.message-container {
  display: flex;
  flex-direction: column;
  gap: 40px;
}

.message-box1 {
  border: 1px solid #e5e5e5;
  border-radius: 3px;
  background-color: #fff;
  height: fit-content;
  width: 35%;
}

.message-chat-right {
  height: 518.2px;
  background-color: #fff;
  border: 1px solid #e5e5e5;
  border-radius: 3px;
}

.m-row {
  display: flex;
  flex-direction: row;
}

/* Chat message list */
.group_messages {
  overflow-y: auto;
  max-height: 444px;
  padding-right: 10px;
}

.chat__message__dt {
  display: flex;
  padding: 10px;
  cursor: pointer;
}

.chat__message__dt.active {
  background-color: #ffecec;
  border-left: 4px solid #ed2a26;
}

.chat__message__dt:hover {
  background-color: #ffecec;
  border-left: 4px solid #ed2a26;
}

.user-status {
  min-height: 44px;
  padding: 0;
  position: relative;
  float: left;
  width: 100%;
}

.user-avatar {
  margin-top: 0px;
  float: left;
  position: relative;
}

.user-avatar img {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: 2px solid #fff;
  box-shadow: 0 1px 4px 4px rgba(0, 0, 0, 0.02) inset;
}

.msg__badge {
  position: absolute;
  bottom: -5px;
  right: -5px;
  background-color: #ff5656;
  color: #fff;
  border-radius: 50%;
  padding: 5px 8px;
  font-size: 12px;
  font-weight: bold;
}

.message-user-status-title {
  color: #333;
  font-size: 16px;
  font-weight: 550;
  font-family: "Roboto", sans-serif;
  line-height: 1.4285714286em;
  text-align: left;
}

.user-status-text {
  color: #686f7a;
  font-size: 14px;
  font-weight: 400;
  font-family: "Roboto", sans-serif;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  margin-right: -110px;
  margin-bottom: 0;
  margin-left: 65px;
}

.eps_dots .dropdown-content {
  display: none;
  position: absolute;
  right: 10px !important;
  top: 30px !important;
  background-color: #fff;
  min-width: 170px !important;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.2);
  z-index: 1;
}

.message-icon-dropdown {
  width: 25px;
  height: 17px;
  margin-top: 3.5px;
}

.message-dropdown-item h4 {
  font-size: 13.5px;
  margin-top: 5px;
  display: flex;
  align-items: center;
  font-weight: normal;
}

.message-dropdown-item {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  gap: 10px;
  padding: 10px;
  color: #686f7a;
}

.message-dropdown-item:hover{
  background-color: #ffecec;
  color: #333;
  cursor: pointer;
}

/* Chatbox */
.chat_header {
  padding: 10px;
  background-color: transparent;
}

.message-name-box {
  display: flex;
  flex-direction: column;
  gap: 5px;
}

.message-user-status-box1 {
  display: flex;
  justify-content: flex-start;
  gap: 20px;
}

.message-user-status-tag-online {
  font-size: 10px;
  width: 50px;
  color: #fff;
  background-color: #4caf50;
  border-radius: 3px;
  display: flex;
  text-transform: uppercase;
  justify-content: center;
}

.message-user-parent {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
}

.message-icon-ellipsis {
  margin: 0 10px 35px 0;
  width: 25px;
  height: 25px;
  color: #afafaf;
}

.message-icon-ellipsis:hover {
  cursor: pointer;
  color: #333;
}

.messages-line {
  height: 370px; /* Adjust this height as needed */
  overflow-y: auto;
  padding: 10px;
  border-top: 1px solid #e5e5e5;
  border-bottom: 1px solid #e5e5e5;
  background-color: transparent;
}

.message-main-message-box-ta-right {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-end;
}

.message-main-message-box-ta-right .message-dt {
  float: right;
}

.main-message-box.st3 {
  text-align: left;
}

.message-dt {
  max-width: 65%;
  position: relative;
  width: -moz-fit-content;
  width: fit-content;
}

.message-inner-dt {
  padding: 15px;
  border-radius: 3px;
  background-color: #ffecec;
  font-size: 14px;
}

.message-main-message-box-ta-right .message-inner-dt {
  background-color: #ed2a26;
}

.message-main-message-box-ta-right .message-inner-dt p {
  color: #fff;
}

.message-main-message-box-ta-right span,
.message-dt.st3 span {
  font-size: 12px;
  color: #686f7a;
  margin-top: 5px;
}

.message-main-message-box-ta-right span {
  float: right;
}

.message-dt.st3 span {
  float: left;
}

.message-send-area {
  padding: 12px;
}

.m-send-message input {
  width: calc(100% - 10px);
  padding: 10px 15px;
  border: 1px solid #e5e5e5;
  font-size: 14px;
  border-radius: 3px;
  color: #333;
  margin-bottom: 0 !important;
}

.m-send-message input::placeholder {
  color: rgb(194, 192, 192);
}

.message-chat {
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.m-field {
  width: 100%;
  display: flex;
}

.m-box-send_message {
  width: 100%;
}

.m-send_button {
  width: 60px;
  height: 38px;
  background-color: #ed2a26; /* Màu đỏ */
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 3px;
  border: none;
  cursor: pointer;
}

.m-send_button:hover {
  background-color: #333;
}

.message-send-icon {
  color: #fff;
  width: 20px;
  height: 20px;
}

.group_messages::-webkit-scrollbar {
  width: 0;
}

.messages-line::-webkit-scrollbar {
  width: 0;
}

@media screen and (max-width: 1050px) {
  .m-row {
    display: block;
  }

  .message-box1 {
    width: 100%;
    margin-bottom: 20px;
  }

  .message-section {
    padding: 0;
  }
}
