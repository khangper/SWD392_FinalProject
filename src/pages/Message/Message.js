import React, { useEffect, useState } from "react";
import "./Message.css";
import { AiOutlineSearch } from "react-icons/ai";
import { PiWechatLogoLight } from "react-icons/pi";
import { VscSend } from "react-icons/vsc";
import { AiOutlineEllipsis } from "react-icons/ai";
import { RiDeleteBin5Line } from "react-icons/ri";
import { MdBlockFlipped } from "react-icons/md";
import { PiFlagCheckered } from "react-icons/pi";
import { BsVolumeMute } from "react-icons/bs";
import { useDispatch, useSelector } from "react-redux";
import { fetchChatMessagesRequest } from "../../redux/reduxActions/MessageAction";

const Message = () => {
  const dispatch = useDispatch();
  const { loading, messages, error } = useSelector((state) => state.message);
  const [selectedIndex, setSelectedIndex] = useState(0);

  const handleItemClick = (index) => {
    setSelectedIndex(index);
  };

  useEffect(() => {
    dispatch(fetchChatMessagesRequest());
  }, [dispatch]);

  if (loading) {
    return <p>Loading...</p>;
  }

  if (error) {
    return <p>Error: {error}</p>;
  }

  return (
    <div>
      <div className="message-section">
        <div className="message-container">
          <div className="message-label">
            <div className="message-titles">
              <div className="message-icon">
                <PiWechatLogoLight className="message-icon-msg" />
              </div>
              <h2 className="message-title">Messages</h2>
            </div>
          </div>

          <div className="m-row">
            <div className="message-box1">
              <div className="message-search">
                <div className="message-box-input-search">
                  <button className="message-button-search">
                    <AiOutlineSearch className="message-icon-search" />
                  </button>
                  <input
                    className="message-search-child"
                    type="text"
                    placeholder="Search Messages..."
                  />
                </div>
              </div>
              <div className="simplebar-content-wrapper">
                <div className="group_messages">
                  {messages.map((message, index) => (
                    <div
                      key={index}
                      className={`chat__message__dt ${
                        selectedIndex === index ? "active" : ""
                      }`}
                      onClick={() => handleItemClick(index)}
                    >
                      <div className="user-status">
                        <div className="user-avatar">
                          <img src={message.image} alt="" />
                          <div className="msg__badge">{message.id}</div>
                        </div>
                        <div className="message-user-name">
                          <div className="message-name-time">
                            <p className="user-status-title">{message.name}</p>
                            <p className="user-status-time floaty">
                              {message.time}
                            </p>
                          </div>
                          <p className="user-status-text">{message.text}</p>
                        </div>
                      </div>
                    </div>
                  ))}
                </div>
              </div>
            </div>
            <div className="message-chat-right">
              <div className="chat_header">
                <div className="message-user-parent">
                  <div className="message-user-status-box1">
                    <div className="user-avatar">
                      <img
                        src="https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-1.jpg"
                        alt=""
                      />
                    </div>
                    <div className="message-name-box">
                      <p className="message-user-status-title">John Doe</p>
                      <p className="message-user-status-tag-online">Online</p>
                    </div>
                  </div>
                  <div className="user-status-time floaty eps_dots eps_dots5 more_dropdown">
                    <AiOutlineEllipsis className="message-icon-ellipsis" />
                    <div className="dropdown-content">
                      <div className="message-dropdown-item">
                        <RiDeleteBin5Line className="message-icon-dropdown" />
                        <h4>Delete</h4>
                      </div>

                      <div className="message-dropdown-item">
                        <MdBlockFlipped className="message-icon-dropdown" />
                        <h4>Block</h4>
                      </div>

                      <div className="message-dropdown-item">
                        <PiFlagCheckered className="message-icon-dropdown" />
                        <h4>Report</h4>
                      </div>

                      <div className="message-dropdown-item">
                        <BsVolumeMute className="message-icon-dropdown" />
                        <h4>Mute</h4>
                      </div>

                      {/* <span>
                        <MdBlockFlipped className="message-icon-dropdown" />
                        Block
                      </span>
                      <span>
                        <PiFlagCheckered className="message-icon-dropdown" />
                        Report
                      </span>
                      <span>
                        <BsVolumeMute className="message-icon-dropdown" />
                        Mute
                      </span> */}
                    </div>
                  </div>
                </div>
              </div>
              <div className="messages-line">
                <div className="message-chat">
                  <div className="message-main-message-box-ta-right">
                    <div className="message-dt">
                      <div className="message-inner-dt">
                        <p>
                          Lorem ipsum dolor sit amet, consectetur adipiscing
                          elit. Donec rutrum congue leo eget malesuada. Vivamus
                          suscipit tortor eget felis porttitor.
                        </p>
                      </div>
                      <span>5 minutes ago</span>
                    </div>
                  </div>
                  <div className="main-message-box st3">
                    <div className="message-dt st3">
                      <div className="message-inner-dt">
                        <p>Cras ultricies ligula.</p>
                      </div>
                      <span>5 minutes ago</span>
                    </div>
                  </div>
                  <div className="message-main-message-box-ta-right">
                    <div className="message-dt">
                      <div className="message-inner-dt">
                        <p>
                          Lorem ipsum dolor sit amet, consectetur adipiscing
                          elit. Donec rutrum congue leo eget malesuada. Vivamus
                          suscipit tortor eget felis porttitor.
                        </p>
                      </div>
                      <span>3 minutes ago</span>
                    </div>
                  </div>
                  <div className="main-message-box st3">
                    <div className="message-dt st3">
                      <div className="message-inner-dt">
                        <p>Lorem ipsum dolor sit amet</p>
                      </div>
                      <span>2 minutes ago</span>
                    </div>
                  </div>
                  <div className="message-main-message-box-ta-right">
                    <div className="message-dt">
                      <div className="message-inner-dt">
                        <p>
                          Lorem ipsum dolor sit amet, consectetur adipiscing
                          elit. Donec rutrum congue leo eget malesuada. Vivamus
                          suscipit tortor eget felis porttitor.
                        </p>
                      </div>
                      <span>1 minutes ago</span>
                    </div>
                  </div>
                  <div className="main-message-box st3">
                    <div className="message-dt st3">
                      <div className="message-inner-dt">
                        <p>....</p>
                      </div>
                      <span>Typing...</span>
                    </div>
                  </div>
                </div>
              </div>
              <div className="message-send-area">
                <form>
                  <div className="m-field">
                    <div className="m-box-send_message">
                      <div className="m-send-message">
                        <input
                          className="prompt srch_explore"
                          type="text"
                          id="chat-widget-message-text-2"
                          name="chat_widget_message_text_2"
                          placeholder="Write a message..."
                        />
                      </div>
                    </div>
                    <button className="m-send_button" type="submit">
                      <VscSend className="message-send-icon" />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Message;
