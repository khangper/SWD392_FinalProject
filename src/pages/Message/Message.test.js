import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import Message from "./Message";
import { screen } from "@testing-library/react";

describe("Message", () => {
    beforeAll(() => {
        jest.spyOn(console, "error").mockImplementation(() => { });
        jest.spyOn(console, "log").mockImplementation(() => { });
    });

    afterAll(() => {
        console.error.mockRestore();
        console.log.mockRestore();
    });
    it("renders the Message component", async () => {
        const { container } = renderWithProviders(<Message />);
        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });
})