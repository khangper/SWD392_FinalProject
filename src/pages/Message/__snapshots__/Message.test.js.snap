// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders the Message component 1`] = `
<div>
  <p>
    Loading...
  </p>
</div>
`;
